### ISSUE TITLE
gzip: invalid option -- 'k' 

### ISSUE THREAD

**Issue by Thunderscreech** [jpoirier/stratux-setup#1]
Board: Raspberry Pi 2
uname -a: Linux raspberrypi 4.4.8-v7+ #880 SMP Fri Apr 22 21:55:04 BST 2016 armv7l GNU/Linux

rpi.sh uses gzip -k which is not supported pre gzip 1.6.  The image I'm installing on has 1.5 with no automagic upgrade path which means someone else out there'll probably run into it too.  Suggestion: check and handle gzip version if -keep functionality is needed.

Steps to reproduce:
Started with an OEM image for an LCD board from https://drive.google.com/file/d/0B5ceUb50sIDnM0hSVDVaU2JVbUk/view?usp=sharing and did full update/upgrade to current kernel.

Followed instructions in readme.MD and the script ran well until:

```
************************************
****** Raspberry Pi setup... *******
************************************


**** Boot config settings... *****
...done

**** Disable serial console... *****
...done

**** RPi 0/2 check to enable Edimax wifi dongle option... *****
copying the hostapd-edimax binary...
gzip: invalid option -- 'k'
Try `gzip --help' for more information.
ERROR - hostapd doesn't exist, exiting...
root@raspberrypi:~/stratux-setup#
```

man gzip showed that k wasn't listed as a supported switch as indicated above.  Because I started with an older version of Raspbian (I think it was Wheezy) and upgraded it, could I have some kind of version issue?  Gzip 1.6 adds the -keep functionality, but apt-cache says I have 1.5 and won't pull a newer version:  

```
root@raspberrypi:~/stratux-setup# apt-cache show gzip
Package: gzip
Version: 1.5-1.1
Architecture: armhf
Essential: yes
Maintainer: Bdale Garbee <bdale@gag.com>
Installed-Size: 224
Pre-Depends: libc6 (>= 2.13-28), libgcc1 (>= 1:4.4.0)
Depends: dpkg (>= 1.15.4) | install-info
Suggests: less
Priority: required
Section: utils
Filename: pool/main/g/gzip/gzip_1.5-1.1_armhf.deb
Size: 115558
SHA256: 76b635907ddbb5c32616f0ec5b6a2b3150e5034e6f60527e3671af27371fd634
SHA1: 3c6515123a1cca364cdc1c4ec65ce64d23687106
MD5sum: 5358035dbb1c5d992fe05999a4b08bb9
Description: GNU compression utilities
```

Might be worth checking which version of gzip is installed, at least one poor sap didn't have the right one and if you're using -k, you might want to handle boxes that can't use that switch.  

hostapd and the DHCP server were both [-] in status and there was no sign of Stratux.

**Comment by jpoirier** [comment]
Thanks for the report! It'd be simpler to default to using the -c option (gunzip -c hostapd.gz >hostapd) rather than check the version. Hmm, not sure why you can't upgrade...

**Comment by Thunderscreech** [comment]
It might be something weird with my OEMified offshoot of Wheezy.  I'm going to just start over with Jessie and see if I can get the LCD hardware to work with it instead, Jessie's got gzip 1.6 so this won't be a problem.  Thought I'd mention it here in case you wanted to know, this switch will bite someone else in the butt if they're trying to use it with Wheezy or before (possibly for same reason I was, poor driver support for weird hardware).

Thanks!

**Comment by jpoirier** [comment]
Feedback is always welcome, thanks!

fixed in #d3196220
